test/parallel_api/iterator/iterators.pass.cpp

```diff
@@ -43,7 +43,9 @@ void test_random_iterator(const RandomIt& it) {
         [[maybe_unused]] auto t4 = typename RandomIt::iterator_category{};
     }
 
-    static_assert(::std::is_default_constructible_v<RandomIt>, "iterator is not default constructible");
```

Comment:
I am not 100% sure this is a safe change to do.

Reply:
As per our offline discussion, I've adjusted the code to have a pair of functions, one which wraps the other:
`test_random_iterator` - does a full check of random access iterator
`test_random_iterator_skip_default_ctor_check` - skips default constructor check

For a `permutation_iterator` based on a lambda index:
  C++17 and earlier, we skip default ctor check. For greater than C++17, we check default ctor.